web-app/js/portal/search/FacetDrilldownPanel.js

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2012 IMOS
+ *
+ * The AODN/IMOS Portal is distributed under the terms of the GNU General Public License
+ *
+ */
+
+Ext.namespace('Portal.search');
+
+Portal.search.FacetDrilldownPanel = Ext.extend(Ext.tree.TreePanel, {
+    constructor: function(cfg) {
+    	cfg = cfg || {};
+    	
+	    Ext.apply(cfg, {
+	        animate: false,
+	        singleExpand: true,
+	        rootVisible: false,
+	        cls: "hierarchicalTree",
+	        lines: false
+	    });
+
+        Portal.search.FacetDrilldownPanel.superclass.constructor.call(this, cfg);
+
+        this._selectedNodeValueHierarchy = "";
+        this._initRootNode();
+
+        this.addEvents('drilldownchange');
+ 
+        this.getSelectionModel().on('selectionchange', this._onSelectionChange, this);
+        this.on('checkchange', this._onCheckChange, this);
+
+        this.mon(this.searcher, 'searchcomplete', function() {
+            this._onSearchComplete();
+        }, this);
+    },
+	
+    getDrilldownPath: function() {
+    	return this._selectedNodeValueHierarchy;
+    },
+
+    _initRootNode: function() {
+    	if (this.searcher.hasFacetNode(this.facetName)) {
+    		this.setRootNode(this.searcher.getFacetNode(this.facetName));
+    	} else {
+    		this.setRootNode(new Ext.tree.TreeNode());
+    	}
+    },
+
+    _onCheckChange: function(node) {
+        this._selectedNodeValueHierarchy = node.parentNode.toValueHierarchy();
+    	this.fireEvent('drilldownchange');
+    },
+
+    _onSelectionChange: function(selectionModel, node) {
+        this._selectedNodeValueHierarchy = node.toValueHierarchy();
+        trackFacetUsage(this.facetName, node.attributes.value);
+        this.fireEvent('drilldownchange');
+    },
+
+    _onSearchComplete: function() {
+    	if (this.searcher.hasFacetNode(this.facetName)) {
+    		this.setRootNode(this.searcher.getFacetNode(this.facetName))
```

Comment:
Looks like my eclipse ide has the wrong setting for indenting for javascript files - I'll fix it and the problems its caused.